docs/src/components/Layout/Sidebar.tsx

```diff
@@ -0,0 +1,299 @@
+  inputComponents,
+  layoutComponents,
+  navigationComponents,
+  utilityComponents,
+  legacyComponents,
+  guides,
+  theming,
+} from '../../data/links';
+
+import Link from 'next/link';
+import { useCustomRouter } from '@/components/useCustomRouter';
+import { FrameworkChooser } from './FrameworkChooser';
+import { LogoLink } from './LogoLink';
+import { MenuButton } from './MenuButton';
+
+const NavLinks = ({
+  items,
+  onClick,
+}: {
+  items: ComponentNavItem[];
+  onClick?: () => void;
+}) => (
+  <Collection type="list" items={items} gap="0">
+    {({ href, label }) => (
+      <NavLink key={label} href={href} onClick={onClick}>
+        {label}
+      </NavLink>
+    )}
+  </Collection>
+);
+
+const NavLink = ({ href, children, onClick, platforms = [] }) => {
+  const {
+    query: { platform = 'react' },
+    asPath,
+  } = useCustomRouter();
+  const isCurrent = asPath === `/${platform}${href}`;
+  const classNames = `docs-secondary-nav-link ${isCurrent ? 'current' : ''}`;
+
+  if (platforms.length && !platforms.includes(platform)) {
+    return null;
+  }
+
+  return (
+    <Link href={`/${platform}${href}`} passHref>
+      <a onClick={onClick} className={classNames}>
+        {children}
+      </a>
+    </Link>
+  );
+};
+
+const NavLinkComponentsSection = ({ heading, components, ...props }) => {
+  const { query } = useCustomRouter();
+  const { tokens } = useTheme();
+  const { platform = 'react' } = query;
+
+  const platformComponents = components.filter((component) => {
+    if (component.platforms) {
+      return component.platforms.includes(platform);
+    }
+    return true;
+  });
+
+  if (!platformComponents.length) {
+    return null;
+  }
+  return (
+    <>
+      <Text
+        fontSize={tokens.fontSizes.xs}
+        fontWeight={tokens.fontWeights.semibold}
+        textTransform="uppercase"
+        letterSpacing="0.125em"
+        color={tokens.colors.font.tertiary}
+        padding={`${tokens.space.small} ${tokens.space.medium} ${tokens.space.xs} var(--secondary-nav-indent)`}
+      >
+        {heading}
+      </Text>
+      <NavLinks {...props} items={platformComponents} />
+    </>
+  );
+};
+
+const ExpanderTitle = ({ Icon, text }) => {
+  const { tokens } = useTheme();
+  return (
+    <Flex
+      direction="row"
+      alignItems="center"
+      gap={tokens.space.small}
+      padding={tokens.space.xs}
+    >
+      <Icon
+        style={{ height: '1.25rem', width: '1.25rem' }}
+        color={tokens.colors.brand.primary[60]}
+      />
+      <Text as="span">{text}</Text>
+    </Flex>
+  );
+};
+
+// TODO: clean up this logic
+const SecondaryNav = (props) => {
+  const {
+    pathname,
+    query: { platform },
+  } = useCustomRouter();
+
+  // Extract section from URL (/section/... => section)
+  const section = pathname.split('/')[2];
+  const [value, setValue] = React.useState<string | string[]>([section]);
+
+  return (
+    <Expander type="multiple" value={value} onChange={setValue}>
```

Comment:
This will make sure the current page's expander item is expanded. Right now because we aren't changing any URLs, all the components pages will expand all component sections (connected, legacy, primitive).